CreeDictionary/API/models.py

```diff
@@ -364,12 +388,13 @@ def search(user_query: str) -> Tuple[str, List[SearchResult]]:
                     part_of_speech=lemma.full_lc,
                     is_lemma=entry.wordform == lemma.text,
                     lemma=lemma.text,
-                    preverbs=(),
-                    reduplication_tags=(),
-                    initial_change_tags=(),
-                    definitions=definitions,
+                    preverbs=(),  # type: ignore
+                    reduplication_tags=(),  # type: ignore
+                    initial_change_tags=(),  # type: ignore
+                    definitions=definitions,  # type: ignore
```

Comment:
If those type: ignores were removed:

```
(cree-intelligent-dictionary) D:\PycharmProjects\cree-intelligent-dictionary>mypy CreeDictionary
CreeDictionary\API\models.py:391: error: Argument "preverbs" to "SearchResult" has incompatible type "Tuple[]"; expected "Tuple[str]"
CreeDictionary\API\models.py:392: error: Argument "reduplication_tags" to "SearchResult" has incompatible type "Tuple[]"; expected "Tuple[str]"
CreeDictionary\API\models.py:393: error: Argument "initial_change_tags" to "SearchResult" has incompatible type "Tuple[]"; expected "Tuple[str]"
CreeDictionary\API\models.py:394: error: Argument "definitions" to "SearchResult" has incompatible type "Tuple[Any, ...]"; expected "Tuple[Definition]"
Found 4 errors in 1 file (checked 64 source files)
```

I think mypy treats empty tuples different from tuples with element annotation. The problem is to tell mypy that the tuple can be empty.

following [this dicussion](https://github.com/python/mypy/issues/4211), I tried to annote the SearchResult class with `Tuple[Union[str, ()]]` and found mypy does not accept that.

Just today, I followed [this suggestion](https://github.com/python/typing/issues/157#issuecomment-162633552) and annotated SearchResult class with `Tuple[str, ...]`. Turns out mypy loves that. I'll push the commit here later.

Reply:
> Just today, I followed [this suggestion](https://github.com/python/typing/issues/157#issuecomment-162633552) and annotated SearchResult class with `Tuple[str, ...]`. Turns out mypy loves that. I'll push the commit here later.

Yeah, we both derped. `Tuple[str, ...]` is the correct type annotation for a tuple with an arbitrary amount of `str`. The reason you can't use `Tuple[str]` is because that type means a tuple of exactly one string :/